### PR TITLE
chore(hc): Adds Debug File upload config endpoint

### PR DESCRIPTION
**Pull request by GabeVillalobos**
Adds a new endpoint that surfaces config information for legacy DSYM file uploads. It currently only surfaces a region URL for a CLI client to upload to the correct region, bypassing the API gateway entirely and avoiding timeout issues. The root cause for the timeout issues is still being determined, but seems to be abnormally long spans originating from the chunk-upload requests being proxied. Regardless, there is no reason for upload traffic to be routed through the API gateway when using the CLI, as it will always incur unnecessary additional latency.

This endpoint requires passing in the organization and project slugs similar to the actual upload endpoint itself in order to pre-construct the full URL for the client.

More info on this change and its potential use case can be found here:
[Region Safe Uploads](https://www.notion.so/sentry/Region-Safe-Uploads-Spec-b431fa85b77e48808ff6eeaf6c61163e?pvs=4)
This is a sibling PR of [getsentry/sentry-cli#1983](https://github.com/getsentry/sentry-cli/pull/1983) which consumes this endpoint and issues DSym uploads to the url provided in the response body.

### CODE REVIEW

File: src/sentry/api/endpoints/debug_files.py [hunk 1]
```diff
@@ -370,6 +373,29 @@ def get(self, request: Request, project) -> Response:
...
+    def get(self, request: Request, project) -> Response:
+        org = request.organization
+        dsym_file_url = reverse(
+            "sentry-api-0-dsym-files",
+            kwargs={"organization_slug": org.slug, "project_slug": project.slug},
+        )
+
+        return Response(
+            {
+                "region_url": absolute_uri(
```
Comment (markstory) [inline src/sentry/api/endpoints/debug_files.py:391]:
```suggestion
                "regionUrl": absolute_uri(
```
most API attributes are camelBacked.

File: src/sentry/api/endpoints/debug_files.py [hunk 2]
```diff
@@ -370,6 +373,29 @@ def get(self, request: Request, project) -> Response:
         return Response({"missing": missing})
 
 
+@region_silo_endpoint
+class DebugFilesConfigEndpoint(ProjectEndpoint):
+    owner = ApiOwner.HYBRID_CLOUD
+    publish_status = {"GET": ApiPublishStatus.PRIVATE}
+    permission_classes = (ProjectReleasePermission,)
+
+    def get(self, request: Request, project) -> Response:
```
Comment (markstory) [inline src/sentry/api/endpoints/debug_files.py:382]: Is the plan to update sentry-cli to use this endpoint to locate the upload URL?

Comment (GabeVillalobos) [reply src/sentry/api/endpoints/debug_files.py:382]: Yep, I'm currently testing the change that consumes this new API in the CLI and uses this for legacy uploads as well. What sucks is that existing CLI users will have to update to fix timeouts/broken uploads. 

As an alternative approach, is there a way we could configure the API proxy to issue a redirect to the correct region vs simply proxying the request?

**Comment by sentaur-athena** [comment]
Can you please add more context about what you're doing and why you have to skip API gateway? What are the trade-offs? Where is this endpoint is going to be used?

**Comment by GabeVillalobos** [comment]
> Can you please add more context about what you're doing and why you have to skip API gateway? What are the trade-offs? Where is this endpoint is going to be used?

Sure thing. Because uploads are currently going through the API Gateway, it's resulting in a large number of RPC calls being made from control-silo to the relevant region which is causing uploads to fail. Because of this, we intend to consume this new endpoint in the CLI implementation + any relevant SDKs with bespoke upload logic if they exist. More details here in our tech-spec for the proposed changes: [Region Safe Uploads](https://www.notion.so/sentry/Region-Safe-Uploads-Spec-b431fa85b77e48808ff6eeaf6c61163e?pvs=4)

File: src/sentry/api/endpoints/debug_files.py (hunk 2 above)
Comment (markstory) [reply src/sentry/api/endpoints/debug_files.py:382]: We might be able to do a redirect. I suspect we have made clients that won't respect redirects though.

**Review by Swatinem — APPROVED** [review]

**Review by markstory — APPROVED** [review]
Looks good to me. Newer clients will absolute URLs. sentry-cli > 1.70.1 and < 2.30.0 wil will continue getting a relative URL.

**Comment by GabeVillalobos** [comment]
I split this change and am closing in favor of #67734

**Pull request closed without merging**